File: MainActivity.java

package com.santik.appsurver;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    EditText phoneInput, amountInput;
    Button sendBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        phoneInput = findViewById(R.id.phoneInput);
        amountInput = findViewById(R.id.amountInput);
        sendBtn = findViewById(R.id.sendBtn);

        sendBtn.setOnClickListener(v -> {
            String phone = phoneInput.getText().toString();
            String amount = amountInput.getText().toString();
            String url = "https://t.me/mobaily_bot?start=" + phone + "_" + amount;
            Intent intent = new Intent(Intent.ACTION_VIEW);
            intent.setData(Uri.parse(url));
            startActivity(intent);
        });
    }
}

File: MainActivity.kt

package com.example.appsurver

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.FirebaseDatabase

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val phoneInput = findViewById<EditText>(R.id.phoneInput)
        val amountInput = findViewById<EditText>(R.id.amountInput)
        val sendButton = findViewById<Button>(R.id.sendButton)

        sendButton.setOnClickListener {
            val phone = phoneInput.text.toString().trim()
            val amount = amountInput.text.toString().trim()

            if (phone.isNotEmpty() && amount.isNotEmpty()) {
                val ref = FirebaseDatabase.getInstance().getReference("flexyRequests")
                val data = mapOf(
                    "phone" to phone,
                    "amount" to amount,
                    "timestamp" to System.currentTimeMillis()
                )
                ref.push().setValue(data)

                val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://t.me/mobaily_bot"))
                startActivity(intent)
            }
        }
    }
}
